omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/NameTemplateController2_0.java

```diff
@@ -0,0 +1,110 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.webservices.rest.web.v1_0.controller.openmrs2_0;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.lang3.StringUtils;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.BaseRestController;
+import org.openmrs.module.webservices.rest.web.v1_0.support.openmrs2_0.NameSupport2_0;
+import org.openmrs.module.webservices.rest.web.v1_0.template.openmrs2_0.NameTemplate2_0;
+import org.openmrs.serialization.SerializationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.context.MessageSource;
+import org.springframework.context.NoSuchMessageException;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.ResponseBody;
+import org.springframework.web.context.request.WebRequest;
+
+@Controller
+@RequestMapping(value = "/rest/" + RestConstants.VERSION_1 + "/nametemplate")
+public class NameTemplateController2_0  extends BaseRestController {
+
+    private static final Logger log = LoggerFactory.getLogger(NameTemplateController2_0.class);
+
+    @RequestMapping(method = RequestMethod.GET)
+    @ResponseBody
+    public Object get(WebRequest request) throws SerializationException {
+
+        NameTemplate2_0 nameTemplate = NameSupport2_0.getInstance().getDefaultLayoutTemplate();
+
+        // Check global properties for defaults/overrides in the form of n=v,n1=v1, etc
+        String customDefaults = Context.getAdministrationService().getGlobalProperty("layout.name.defaults");
+        if (customDefaults != null) {
+            String[] tokens = customDefaults.split(",");
+            Map<String, String> elementDefaults = nameTemplate.getElementDefaults();
+
+            for (String token : tokens) {
+                String[] pair = token.split("=");
+                if (pair.length == 2) {
+                    String name = pair[0];
+                    String val = pair[1];
+
+                    if (elementDefaults == null) {
```

Comment:
you wanna do something similar here

`if ( StringUtils.isEmpty(elementDefaults)) {`